planutils/__init__.py

```diff
@@ -1,31 +1,47 @@
 
 import argparse, os
 
+from planutils import settings
+
 
 def setup():
 
-    assert not_setup_yet(), "Error: planutils is already setup. Remove ~/.planutils to reset (warning: all cached planners will be lost)."
+    assert not_setup_yet(), "Error: planutils is already setup. Remove ~/.planutils to reset (warning: all cached packages will be lost)."
+
+    CUR_DIR = os.path.dirname(os.path.abspath(__file__))
 
     print("\nCreating ~/.planutils...")
     os.mkdir(os.path.join(os.path.expanduser('~'), '.planutils'))
     os.mkdir(os.path.join(os.path.expanduser('~'), '.planutils', 'bin'))
-    os.mkdir(os.path.join(os.path.expanduser('~'), '.planutils', 'bin', 'images'))
+
+    settings.save({
+        'installed': []
+    })
+
+    os.symlink(os.path.join(CUR_DIR, 'packages'),
+               os.path.join(os.path.expanduser('~'), '.planutils', 'packages'))
 
     print("Adding bin folder to path (assuming ~/.bashrc exists)...")
-    os.system('echo \'export PATH="$HOME/.planutils/bin:$PATH"\' >> ~/.bashrc')
+    os.system("echo 'export PLANUTILS_PREFIX=\"~/.planutils\"' >> ~/.bashrc")
+    os.system("echo 'export PATH=\"$PLANUTILS_PREFIX/bin:$PATH\"' >> ~/.bashrc")
 
-    print("Installing planner scripts...")
-    from planutils.planner_installation import PLANNERS
-    for p in PLANNERS:
+    print("Installing package scripts...")
+    from planutils.package_installation import PACKAGES
+    for p in PACKAGES:
         script  = "#!/bin/bash\n"
-        script += "echo\n"
-        script += "echo 'Planner not installed!'\n"
-        script += "read -p \"Download & install? [y/n] \" varchoice\n"
-        script += "if [ $varchoice == \"y\" ]\n"
+        script += "if [ \"$(planutils --check_installed %s)\" == \"True\" ]\n" % p
         script += "then\n"
-        script += "  planutils --install " + p + "\n"
-        script += "fi\n"
-        script += "echo"
+        script += "  ~/.planutils/packages/%s/run $@\n" % p
+        script += "else\n"
+        script += "  echo\n"
+        script += "  echo 'Package not installed!'\n"
+        script += "  read -p \"Download & install? [y/n] \" varchoice\n"
+        script += "  if [ $varchoice == \"y\" ]\n"
+        script += "  then\n"
+        script += "    planutils --install " + p + "\n"
+        script += "  fi\n"
```

Comment:
Since we still have the parameters that the user wanted to run at this point, we could ask: "Successfully installed %s. Do you want to run your original command now?". Maybe print the command line as well and then have a yes/no dialog as above.